src/workflows-app/utils/submission-utils.ts

```diff
@@ -477,3 +500,66 @@ export const validateInputs = (
     const inputMessage = validateInput(input, dataTableAttributes);
     return { name: 'input_name' in input ? input.input_name : input.field_name, ...inputMessage };
   });
+
+export const getInputTableData = (
+  configuredInputDefinition: InputDefinition[],
+  searchFilter: string,
+  includeOptionalInputs: boolean,
+  inputTableSort: { field: string; direction: boolean | 'asc' | 'desc' }
+): InputTableData[] => {
+  const val = _.flow(
+    (rows: InputDefinition[]) =>
+      rows.map((row, index): InputTableData => {
+        const { workflow, call, variable } = parseMethodString(row.input_name);
+        return {
+          taskName: call || workflow || '',
+          variable: variable || '',
+          inputTypeStr: renderTypeText(row.input_type),
+          configurationIndex: index,
+          optional: isInputOptional(row.input_type),
+          ...row,
+        };
+      }),
+    _.orderBy<InputTableData>(
+      [
+        'optional',
+        // @ts-expect-error
+        ({ [inputTableSort.field]: field }) => _.lowerCase(field),
+        ({ taskName }) => _.lowerCase(taskName),
+        ({ variable }) => _.lowerCase(variable),
+      ],
+      ['asc', inputTableSort.direction, 'asc', 'asc']
+    ),
+    (rows: InputTableData[]) => {
+      return rows.filter((row: InputTableData) => {
+        return (
+          (includeOptionalInputs || !row.optional) &&
+          (row.taskName.toLocaleLowerCase().includes(searchFilter.toLocaleLowerCase()) ||
+            row.variable.toLocaleLowerCase().includes(searchFilter.toLocaleLowerCase()))
+        );
+      });
+    }
+  )(configuredInputDefinition);
+
+  return val;
+};
+
+export const getOutputTableData = (
+  configuredOutputDefinition: OutputDefinition[],
+  sort: { field: string; direction: string }
+): OutputTableData[] => {
+  return _.flow(
+    _.entries,
```

Comment:
`toIndexPairs` handles converting the index to a number, which removes the need for `parseInt` when setting the configuration index.

```suggestion
    Utils.toIndexPairs,
```